### ISSUE TITLE
Support OpenSSL 3.0

### ISSUE THREAD

**Issue by pantharshit00** [prisma/prisma#11356]
### Discussed in https://github.com/prisma/prisma/discussions/11345

<div type='discussions-op-text'>

<sup>Originally posted by **HKGx** January 24, 2022</sup>
OpenSSL 3.0 has been released in September and probably is pretty much backwards compatible with OpenSSL 1.1.1. Are there any plans for supporting it?
https://www.openssl.org/blog/blog/2021/09/07/OpenSSL3.Final/
</div>

**Comment by janpio** [comment]
Current error messages are visible here: https://github.com/prisma/prisma/issues/11677#issue-1126136194

**Comment by bopucky** [comment]
I am also receiving the same error, on Ubuntu 22.04 (dev)

`$ npx prisma generate
Error: Unknown binaryTarget debian-openssl-3.0.x and no custom binaries were provided`

OpenSSL version is 3.0.1 14

`$ openssl version
OpenSSL 3.0.1 14 Dec 2021 (Library: OpenSSL 3.0.1 14 Dec 2021)
`

Executing "yarn dev" produces 

`Server is running on http://localhost:4000/
/home/vee/Dev/graphql/hackernews-react-apollo/server/node_modules/@prisma/client/runtime/index.js:36111
      throw new PrismaClientInitializationError(`Unknown ${import_chalk7.default.red("PRISMA_QUERY_ENGINE_LIBRARY")} ${import_chalk7.default.redBright.bold(this.platform)}. Possible binaryTargets: ${import_chalk7.default.greenBright(knownPlatforms.join(", "))} or a path to the query engine library.
            ^

PrismaClientInitializationError: Unknown PRISMA_QUERY_ENGINE_LIBRARY undefined. Possible binaryTargets: darwin, darwin-arm64, debian-openssl-1.0.x, debian-openssl-1.1.x, rhel-openssl-1.0.x, rhel-openssl-1.1.x, linux-arm64-openssl-1.1.x, linux-arm64-openssl-1.0.x, linux-arm-openssl-1.1.x, linux-arm-openssl-1.0.x, linux-musl, linux-nixos, windows, freebsd11, freebsd12, openbsd, netbsd, arm, native or a path to the query engine library.
You may have to run prisma generate for your changes to take effect.
    at Object.getPlatform (/home/vee/Dev/graphql/hackernews-react-apollo/server/node_modules/@prisma/client/runtime/index.js:36111:13)
    at async Object.instantiateLibrary (/home/vee/Dev/graphql/hackernews-react-apollo/server/node_modules/@prisma/client/runtime/index.js:36102:21) {
  clientVersion: '3.8.1',
  errorCode: undefined
}
[nodemon] app crashed - waiting for file changes before starting...
`

**Comment by newtykins** [comment]
I am struggling with the same issue.

![image](https://user-images.githubusercontent.com/19301186/156885605-d35428fe-f6ab-40ca-81f0-39df6546b2c0.png)

![image](https://user-images.githubusercontent.com/19301186/156885611-fceb45a1-0431-4806-b638-9490297bcf70.png)

**Comment by hashsaksham** [comment]
Hey, is there any way I can solve this problem (even temporarily)? I recently updated to Ubuntu 22.04. 
I really need to submit a project. 
Getting the exact same problem while using prisma.

**Comment by vincentbavitz** [comment]
Would love to see a fix here. Recently upgraded to Fedora 36 and getting this issue. Really don't want to downgrade the entire system just for prisma.

**Comment by asaadmohammed74** [comment]
> Would love to see a fix here. Recently upgraded to Fedora 36 and getting this issue. Really don't want to downgrade the entire system just for prisma.

i have the same problem, also on fedora 36.

a quick patch to get things moving would be to patch `getOpenSSLVersion` to return '1.1.x', also you should know i didn't  extensively test this.

```
+++ ./node_modules/prisma/build/index.js
@@ -12938,6 +12938,8 @@ var require_getPlatform = __commonJS({
     __name32(parseOpenSSLVersion, "parseOpenSSLVersion");
     exports2.parseOpenSSLVersion = parseOpenSSLVersion;
     async function getOpenSSLVersion() {
+      return '1.1.x'
+
       const [version2, ls] = await Promise.all([
         gracefulExec(`openssl version -v`),
         gracefulExec(`

```

**Comment by janpio** [comment]
Internal: Also affects Migrations, Introspection and Fromatter Engine.

**Comment by vincentbavitz** [comment]
The best solution I've come up with so far is to run a VM with a shared folder of the build directory and run all prisma commands from the host's CLI. Eg with VirtualBox,
`VBoxManage guestcontrol "MyVM" run -v --username root --password # -- "cd /path/to/project && yarn prisma generate"`

But this is far from ideal...

**Comment by GauravRajSharma** [comment]
As Ubuntu 22.04 is releasing soon, open SSL 3 is the default. Really waiting for this issue to be resolved.

**Comment by vincentbavitz** [comment]
I'd be willing to pay a juicy bounty if we can get this issue moving. This one thing is holding up our entire company and we just need a one-liner fix from Prisma. 🙏🏻

**Comment by janpio** [comment]
This will be tackled soon and fixed. But of course for us this is not close to a one liner fix, as it requires additonal build images, CI runs, image uploads, and actually probably a 3 line change in our download logic 😆 But it is in the works.

**Comment by vincentbavitz** [comment]
@janpio Understandable. Thanks for taking on the issue bro.

**Comment by ranjan-purbey** [comment]
We can probably allow the user to optionally specify platform using an ENV var here: 
https://github.com/prisma/engines-wrapper/blob/759f13aa2841e94bcc35d656e4ec7f56b9f33c50/packages/get-platform/src/getPlatform.ts#L175

**Comment by berningo** [comment]
> > Would love to see a fix here. Recently upgraded to Fedora 36 and getting this issue. Really don't want to downgrade the entire system just for prisma.
> 
> i have the same problem, also on fedora 36.
> 
> a quick patch to get things moving would be to patch `getOpenSSLVersion` to return '1.1.x', also you should know i didn't extensively test this.
> 
> ```
> +++ ./node_modules/prisma/build/index.js
> @@ -12938,6 +12938,8 @@ var require_getPlatform = __commonJS({
>      __name32(parseOpenSSLVersion, "parseOpenSSLVersion");
>      exports2.parseOpenSSLVersion = parseOpenSSLVersion;
>      async function getOpenSSLVersion() {
> +      return '1.1.x'
> +
>        const [version2, ls] = await Promise.all([
>          gracefulExec(`openssl version -v`),
>          gracefulExec(`
> ```

Thank you very much for the hint. How I solved the problem in Ubuntu 22.04 "Jammy Jellyfish":

- Add this `return '1.1.x';` as per quotation after the function definition in each of the following file (search for *all occurrences* of `async function getOpenSSLVersion() {` within the same file):
    * node_modules/prisma/build/index.js
    * node_modules/prisma/prisma-client/runtime/index.js
    * node_modules/prisma/prisma-client/generator-build/index.js
    * node_modules/@prisma/engines/dist/index.js
- Download the [latest libssl1.1 binaries from here](https://answers.launchpad.net/ubuntu/jammy/+package/libssl1.1)
- Install as usual, e.g. `sudo apt install ./libssl1.1_1.1.1l-1ubuntu1_amd64.deb`

That solved my problem with prisma 3.7.0:
```
$ yarn prisma --version                                                                                 130 ↵
yarn run v1.22.18
$ prisma --version
Environment variables loaded from .env
prisma                  : 3.7.0
@prisma/client          : 3.7.0
Current platform        : debian-openssl-1.1.x
Query Engine (Node-API) : libquery-engine 8746e055198f517658c08a0c426c7eec87f5a85f (at node_modules/@prisma/engines/libquery_engine-debian-openssl-1.1.x.so.node)
Migration Engine        : migration-engine-cli 8746e055198f517658c08a0c426c7eec87f5a85f (at node_modules/@prisma/engines/migration-engine-debian-openssl-1.1.x)
Introspection Engine    : introspection-core 8746e055198f517658c08a0c426c7eec87f5a85f (at node_modules/@prisma/engines/introspection-engine-debian-openssl-1.1.x)
Format Binary           : prisma-fmt 8746e055198f517658c08a0c426c7eec87f5a85f (at node_modules/@prisma/engines/prisma-fmt-debian-openssl-1.1.x)
Default Engines Hash    : 8746e055198f517658c08a0c426c7eec87f5a85f
Studio                  : 0.445.0
Preview Features        : fullTextSearch
┌─────────────────────────────────────────────────────────┐
│  Update available 3.7.0 -> 3.12.0                       │
│  Run the following to update                            │
│    yarn add --dev prisma@latest                         │
│    yarn add @prisma/client@latest                       │
└─────────────────────────────────────────────────────────┘
Done in 0.72s.
```

```
$ neofetch
            .-/+oossssoo+/-.               oliver@Tuxedo 
        `:+ssssssssssssssssss+:`           ------------- 
      -+ssssssssssssssssssyyssss+-         OS: Ubuntu Jammy Jellyfish (development branch) x86_64 
    .ossssssssssssssssssdMMMNysssso.       Host: TUXEDO InfinityBook S 15 Gen6 
   /ssssssssssshdmmNNmmyNMMMMhssssss/      Kernel: 5.15.0-25-generic 
  +ssssssssshmydMMMMMMMNddddyssssssss+     Uptime: 3 hours, 2 mins 
 /sssssssshNMMMyhhyyyyhmNMMMNhssssssss/    Packages: 1866 (dpkg), 17 (snap) 
.ssssssssdMMMNhsssssssssshNMMMdssssssss.   Shell: zsh 5.8.1 
+sssshhhyNMMNyssssssssssssyNMMMysssssss+   Resolution: 1920x1080 
ossyNMMMNyMMhsssssssssssssshmmmhssssssso   DE: GNOME 42.0 
ossyNMMMNyMMhsssssssssssssshmmmhssssssso   WM: Mutter 
+sssshhhyNMMNyssssssssssssyNMMMysssssss+   WM Theme: Adwaita 
.ssssssssdMMMNhsssssssssshNMMMdssssssss.   Theme: Yaru [GTK2/3] 
 /sssssssshNMMMyhhyyyyhdNMMMNhssssssss/    Icons: Yaru [GTK2/3] 
  +sssssssssdmydMMMMMMMMddddyssssssss+     Terminal: gnome-terminal 
   /ssssssssssshdmNNNNmyNMMMMhssssss/      CPU: 11th Gen Intel i5-1135G7 (8) @ 4.200GHz 
    .ossssssssssssssssssdMMMNysssso.       GPU: Intel TigerLake-LP GT2 [Iris Xe Graphics] 
      -+sssssssssssssssssyyyssss+-         Memory: 5898MiB / 31901MiB 
        `:+ssssssssssssssssss+:`
            .-/+oossssoo+/-.                                       
                                                                   
```

**Comment by dpetrick** [comment]
OpenSSL 3.0.x compatible binaries will be available with 3.13.

**Comment by vincentbavitz** [comment]
@dpetrick @janpio THANKYOU GUYS!

**Comment by diwu1989** [comment]
How do I make a private early release build with openssl 3 support?

**Comment by mrdenysu** [comment]
When is the release of version 3.13?

**Comment by vincentbavitz** [comment]
26 April in the best case ^ @mrdenysu

**Comment by janpio** [comment]
3.13.0 will be released next Tuesday.

You can use `prisma@dev` and `@prisma/client@dev` until then, which is our development version that will be stabilized next week. Please note that it includes other changes as well that are not documented yet (besides the commit history in this repository and the prisma-engines repository).

**Comment by ParkerRoR** [comment]
i has same error, i just typed "sudo apt-get autoremove openssl" in my terminal and all can back running with prisma

**Comment by nowlena** [comment]
I have the following dependencies installed:

 - dependencies: `"@prisma/client": "^3.13.0-dev.29"`
 - devDependencies: `"prisma": "^3.13.0-dev.29"`

When I run `npx prisma@dev generate` I get the following message which appears to be causing errors as the generated client is referencing an older version

```sh
Generated Prisma Client (3.13.0-integration-report-failed-message-is-never-displayed.1 | library) to ./node_modules/@prisma/client in 85ms
You can now start using Prisma Client in your code. Reference: https://pris.ly/d/client

import { PrismaClient } from '@prisma/client'
const prisma = new PrismaClient()

warn Versions of prisma@3.13.0-dev.29 and @prisma/client@3.13.0-integration-report-failed-message-is-never-displayed.1 don't match.
This might lead to unexpected behavior.
Please make sure they have the same version.
```

The actual error I receive when trying to query my DB is `Unknown PRISMA_QUERY_ENGINE_LIBRARY debian-openssl-3.0.x` where I am asked to run `prisma generate` again.

**Comment by janpio** [comment]
@nowlena The output indicates your project actually has a different version installed than what is defined in your ´package.json` - please double check and if necessary nuke your `node_modules` and run your install command again (`npm install` for example). The query error would most probably be a follow up error of that.

**Comment by nowlena** [comment]
> @nowlena The output indicates your project actually has a different version installed than what is defined in your ´package.json`- please double check and if necessary nuke your`node_modules` and run your install command again (`npm install` for example). The query error would most probably be a follow up error of that.

Thanks for the quick reply! I tried deleting my `node_modules` and `package-lock.json` previously.

It appears that the fix was removing the `^` from my dependencies... so now my deps are as follows and everything works!!

 - dependencies: `"@prisma/client": "3.13.0-dev.29"`
 - devDependencies: `"prisma": "3.13.0-dev.29"`

**Comment by janpio** [comment]
Ah sorry, that npm weirdness again. It does not like the way we use versions for our `dev` or `integration` versions.

**Comment by mathefeitosa** [comment]
> 



> > > Would love to see a fix here. Recently upgraded to Fedora 36 and getting this issue. Really don't want to downgrade the entire system just for prisma.
> > 
> > 
> > i have the same problem, also on fedora 36.
> > a quick patch to get things moving would be to patch `getOpenSSLVersion` to return '1.1.x', also you should know i didn't extensively test this.
> > ```
> > +++ ./node_modules/prisma/build/index.js
> > @@ -12938,6 +12938,8 @@ var require_getPlatform = __commonJS({
> >      __name32(parseOpenSSLVersion, "parseOpenSSLVersion");
> >      exports2.parseOpenSSLVersion = parseOpenSSLVersion;
> >      async function getOpenSSLVersion() {
> > +      return '1.1.x'
> > +
> >        const [version2, ls] = await Promise.all([
> >          gracefulExec(`openssl version -v`),
> >          gracefulExec(`
> > ```
> 
> Thank you very much for the hint. How I solved the problem with `yarn prisma migrate dev` in Ubuntu 22.04 "Jammy Jellyfish":
> 
> * Add this `return '1.1.x';` as per quotation after the function definition in each of the following file (search for _all occurrences_ of `async function getOpenSSLVersion() {` within the same file):
>   
>   * node_modules/prisma/build/index.js
>   * node_modules/prisma/prisma-client/runtime/index.js
>   * node_modules/prisma/prisma-client/generator-build/index.js
>   * node_modules/@prisma/engines/dist/index.js
> * Download the [latest libssl1.1 binaries from here](https://answers.launchpad.net/ubuntu/jammy/+package/libssl1.1)
> * Install as usual, e.g. `sudo apt install ./libssl1.1_1.1.1l-1ubuntu1_amd64.deb`
> 
> That solved my problem with prisma 3.7.0:
> 
> ```
> $ yarn prisma --version                                                                                 130 ↵
> yarn run v1.22.18
> $ prisma --version
> Environment variables loaded from .env
> prisma                  : 3.7.0
> @prisma/client          : 3.7.0
> Current platform        : debian-openssl-1.1.x
> Query Engine (Node-API) : libquery-engine 8746e055198f517658c08a0c426c7eec87f5a85f (at node_modules/@prisma/engines/libquery_engine-debian-openssl-1.1.x.so.node)
> Migration Engine        : migration-engine-cli 8746e055198f517658c08a0c426c7eec87f5a85f (at node_modules/@prisma/engines/migration-engine-debian-openssl-1.1.x)
> Introspection Engine    : introspection-core 8746e055198f517658c08a0c426c7eec87f5a85f (at node_modules/@prisma/engines/introspection-engine-debian-openssl-1.1.x)
> Format Binary           : prisma-fmt 8746e055198f517658c08a0c426c7eec87f5a85f (at node_modules/@prisma/engines/prisma-fmt-debian-openssl-1.1.x)
> Default Engines Hash    : 8746e055198f517658c08a0c426c7eec87f5a85f
> Studio                  : 0.445.0
> Preview Features        : fullTextSearch
> ┌─────────────────────────────────────────────────────────┐
> │  Update available 3.7.0 -> 3.12.0                       │
> │  Run the following to update                            │
> │    yarn add --dev prisma@latest                         │
> │    yarn add @prisma/client@latest                       │
> └─────────────────────────────────────────────────────────┘
> Done in 0.72s.
> ```
> 
> The workaround maybe needs to be repeated.

Very good! It worked for me on Ubuntu 22; installing the openssl1.1 folowing the tutorial below.
[Installing OpenSSL 1.1Tutorial Link](https://fedingo.com/how-to-install-openssl-in-ubuntu/)

**Comment by FelipeSantos92Dev** [comment]
> 3.13.0 will be released next Tuesday.
> 
> You can use `prisma@dev` and `@prisma/client@dev` until then, which is our development version that will be stabilized next week. Please note that it includes other changes as well that are not documented yet (besides the commit history in this repository and the prisma-engines repository).

Thank you! It work for me!

**Comment by pemasat** [comment]
> 

Unfurnatly not help. When I tried run
```
npx prisma@dev generate
```
it crashed on

```
Environment variables loaded from .env
Prisma schema loaded from prisma/schema.prisma
> Downloading Prisma engines for Node-API for debian-openssl-3.0.x [                    ] 0%Error: Not Found https://binaries.prisma.sh/all_commits/22b822189f46ef0dc5c5b503368d1bee01213980/debian-openssl-3.0.x/libquery_engine.so.node.gz
```

System Kubuntu 22.04.

**Comment by janpio** [comment]
That is... unexpected 🤔 

But 3.13.0 is out as a normal release now as well. Can you please try to use that one directly?

**Comment by pemasat** [comment]
> That is... unexpected thinking
> 
> But 3.13.0 is out as a normal release now as well. Can you please try to use that one directly?
```
npx prisma generate
```
---> 
```
Error: Unknown binaryTarget debian-openssl-3.0.x and no custom binaries were provided
```
: (

**Comment by marcelorbueno** [comment]
Dears,

* Running on Ubuntu 22.04;
* openssl version:
OpenSSL 3.0.2 15 Mar 2022 (Library: OpenSSL 3.0.2 15 Mar 2022)

I set my `schema.prisma` like:
(...)
generator client {
  provider      = "prisma-client-js"
  **binaryTargets = ["debian-openssl-3.0.x"]**
}
(...)

And this worked fine! 😊🚀❤️

**Comment by janpio** [comment]
> **binaryTargets = ["debian-openssl-3.0.x"]**

Should not even be necessary - if it is, please open a new issue.

**Comment by prosperoking** [comment]
I get this error: Error: 
> Not Found https://binaries.prisma.sh/all_commits/22b822189f46ef0dc5c5b503368d1bee01213980/debian-openssl-3.0.x/libquery_engine.so.node.gz

**Comment by Vitorrrocha** [comment]
Like @marcelorbueno sad above.
I had this same problem when I update my **PopOS to 22.04** and I fix with the code bellow:
At `schema.prisma` file I added in the _generator cliente_ my current debian-openssl version _(you can also add more than one like I did!)_ :
```
generator client {
  provider      = "prisma-client-js"
  binaryTargets = ["native", "debian-openssl-1.1.x", "debian-openssl-3.0.x"]
}
```
and after that I just run the `prisma generate`, if this don't work and show the error message **prisma: command not found** , try to run: `npx prisma generate`

**Comment by prosperoking** [comment]
@Vitorrrocha unfortunately this did not work for me.  still returns the same message running prisma 3.13

**Comment by prosperoking** [comment]
so, i found out i did not update @prisma/client

**Comment by psdkumar** [comment]
Upgrading the libraries `prisma` and `@prisma/client` to `3.13.0` version resolved the issue for me.

**Comment by gBusato** [comment]
> > 
> 
> > > > Would love to see a fix here. Recently upgraded to Fedora 36 and getting this issue. Really don't want to downgrade the entire system just for prisma.
> > > 
> > > 
> > > i have the same problem, also on fedora 36.
> > > a quick patch to get things moving would be to patch `getOpenSSLVersion` to return '1.1.x', also you should know i didn't extensively test this.
> > > ```
> > > +++ ./node_modules/prisma/build/index.js
> > > @@ -12938,6 +12938,8 @@ var require_getPlatform = __commonJS({
> > >      __name32(parseOpenSSLVersion, "parseOpenSSLVersion");
> > >      exports2.parseOpenSSLVersion = parseOpenSSLVersion;
> > >      async function getOpenSSLVersion() {
> > > +      return '1.1.x'
> > > +
> > >        const [version2, ls] = await Promise.all([
> > >          gracefulExec(`openssl version -v`),
> > >          gracefulExec(`
> > > ```
> > 
> > 
> > Thank you very much for the hint. How I solved the problem with `yarn prisma migrate dev` in Ubuntu 22.04 "Jammy Jellyfish":
> > 
> > * Add this `return '1.1.x';` as per quotation after the function definition in each of the following file (search for _all occurrences_ of `async function getOpenSSLVersion() {` within the same file):
> >   
> >   * node_modules/prisma/build/index.js
> >   * node_modules/prisma/prisma-client/runtime/index.js
> >   * node_modules/prisma/prisma-client/generator-build/index.js
> >   * node_modules/@prisma/engines/dist/index.js
> > * Download the [latest libssl1.1 binaries from here](https://answers.launchpad.net/ubuntu/jammy/+package/libssl1.1)
> > * Install as usual, e.g. `sudo apt install ./libssl1.1_1.1.1l-1ubuntu1_amd64.deb`
> > 
> > That solved my problem with prisma 3.7.0:
> > ```
> > $ yarn prisma --version                                                                                 130 ↵
> > yarn run v1.22.18
> > $ prisma --version
> > Environment variables loaded from .env
> > prisma                  : 3.7.0
> > @prisma/client          : 3.7.0
> > Current platform        : debian-openssl-1.1.x
> > Query Engine (Node-API) : libquery-engine 8746e055198f517658c08a0c426c7eec87f5a85f (at node_modules/@prisma/engines/libquery_engine-debian-openssl-1.1.x.so.node)
> > Migration Engine        : migration-engine-cli 8746e055198f517658c08a0c426c7eec87f5a85f (at node_modules/@prisma/engines/migration-engine-debian-openssl-1.1.x)
> > Introspection Engine    : introspection-core 8746e055198f517658c08a0c426c7eec87f5a85f (at node_modules/@prisma/engines/introspection-engine-debian-openssl-1.1.x)
> > Format Binary           : prisma-fmt 8746e055198f517658c08a0c426c7eec87f5a85f (at node_modules/@prisma/engines/prisma-fmt-debian-openssl-1.1.x)
> > Default Engines Hash    : 8746e055198f517658c08a0c426c7eec87f5a85f
> > Studio                  : 0.445.0
> > Preview Features        : fullTextSearch
> > ┌─────────────────────────────────────────────────────────┐
> > │  Update available 3.7.0 -> 3.12.0                       │
> > │  Run the following to update                            │
> > │    yarn add --dev prisma@latest                         │
> > │    yarn add @prisma/client@latest                       │
> > └─────────────────────────────────────────────────────────┘
> > Done in 0.72s.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > The workaround maybe needs to be repeated.
> 
> Very good! It worked for me on Ubuntu 22; installing the openssl1.1 folowing the tutorial below. [Installing OpenSSL 1.1Tutorial Link](https://fedingo.com/how-to-install-openssl-in-ubuntu/)

Thanks for sharing the tutorial link, it helps me to fixe the issue !

**Comment by code33** [comment]
> > @nowlena The output indicates your project actually has a different version installed than what is defined in your ´package.json`- please double check and if necessary nuke your`node_modules` and run your install command again (`npm install` for example). The query error would most probably be a follow up error of that.
> 
> Thanks for the quick reply! I tried deleting my `node_modules` and `package-lock.json` previously.
> 
> It appears that the fix was removing the `^` from my dependencies... so now my deps are as follows and everything works!!
> 
> * dependencies: `"@prisma/client": "3.13.0-dev.29"`
> * devDependencies: `"prisma": "3.13.0-dev.29"`

thanks a lot , perfect

**Comment by ridhwaans** [comment]
it seems like this error is back. Is anyone else having the same issue?
Running OpenSSL 3.0.2 on 22.04 WSL  
Tried with `binaryTargets = ["native", "debian-openssl-1.1.x", "debian-openssl-3.0.x"]` on & off
Tried with prisma & @prisma/client versions `3.13.0-dev.29`, `3.13`, `^3.14`, `3.14`, ran `npx prisma generate`
Still experiencing `Error: Unknown binaryTarget debian-openssl-3.0.x and no custom binaries were provided`
Feedback appreciated

**Comment by tbillington** [comment]
Do you have a position on backporting this to prisma 2?

**Comment by janpio** [comment]
Yes, we generally don't. Upgrading Prisma is pretty simple usually and generally desirable.

**Comment by tbillington** [comment]
That's fair, appreciate the quick response @janpio :)

Just for context, the 2 -> 3 of prisma itself is relatively minor but the ecosystem churn is the pain point. We used `nexus-plugin-prisma` to expose prisma types over graphql, and the migration path for prisma 3 is non-trivial.

**Comment by vntech-engineering93** [comment]
Had this issue since I was working on an older repo + Ubuntu 22.04, updating from 4.3.1 to 4.10.0 solved mine.

**Comment by cesarvspr** [comment]
Changing the `getOpenSSLVersion()` does not work for me. I am using prisma 2.26 and unable to upgrade atm.

OS is fedora 39 and OpenSSL 3.1.1
